app/components/lock_old_posts.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+import datetime as dt
+from itertools import islice
+
+import discord
+
+from app import config
+from app.utils import dynamic_timestamp, post_is_solved
+
+
+async def check_for_old_posts(message: discord.Message) -> None:
+    post = message.channel
+    now = dt.datetime.now(tz=dt.UTC)
+    if (
+        not isinstance(post, discord.Thread)
+        or not post.parent
+        or post.parent.id != config.HELP_CHANNEL_ID
+        or post.locked
+        or post.last_message_id is None
+        or not post_is_solved(post)
+    ):
+        return
+    try:
+        # Ignore messages less than a minute old for the same person may have
+        # sent multiple in quick succession, resulting in their own message
+        # being considered as activity.
+        last = await _get_message(post, 1, before=now - dt.timedelta(minutes=1))
+    except IndexError:
+        # There were no messages older than one minute. Assuming the threshold
+        # below stays at a reasonably large limit, this either means that every
+        # message was deleted, or that there was no response and the original
+        # poster deleted the starter message. It's very unlikely for the post
+        # to no longer be relevant; nor is it likely that anyone would even
+        # message in such a thread. Thus, silently exit if this is the case.
+        return
+    if last.created_at > now - dt.timedelta(seconds=30):
```

Comment:
Oops I forgot to change this back to days, leaving this here so I don't forget again.
```suggestion
    if last.created_at > now - dt.timedelta(days=30):
```